src/commonComponents/sidebar/helpers/sidebar.general.js

```diff
@@ -0,0 +1,18 @@
+// helpers
+import { sanitizeText } from "../../../utility/sanitizeText";
+
+export const getMatchedIndexWithUrlHash = (links) => {
+  const {
+    location: { hash },
+  } = window;
+
+  for (const [idx, link] of links.entries()) {
+    const sanitizedLink = sanitizeText(link);
+
+    if (hash === `#${sanitizedLink}`) {
```

Comment:
If we have a hash which is not an id in our links then it will highlight the first category so once check this